controllers/uploadpackage.lua

```diff
@@ -107,7 +107,7 @@ return {
 
     -- save the file to disk. function defined above.
     save_file(self)
-    local url = string.format("data/%s/%s/%s/%s", self.session.name, self.params.name, self.params.version, filename)
+    local url = string.format(self.config.base_url .. "data/%s/%s/%s/%s", self.session.name, self.params.name, self.params.version, filename)
```

Comment:
I'm fine with this, but maybe include it as "%sdata/%s/%s/%s/%s" and adjust the parameters rather than performing concatenation inside a string.format call?